govcd/vdc.go

```diff
@@ -1350,3 +1353,62 @@ func (vdc *Vdc) CloneVapp(sourceVapp *types.CloneVAppParams) (*VApp, error) {
 	err = vapp.Refresh()
 	return vapp, err
 }
+
+// Get the details of a hardware version
+func (vdc *Vdc) GetHardwareVersion(name string) (*types.VirtualHardwareVersion, error) {
+	found := false
+	for _, hwVersion := range vdc.Vdc.Capabilities[0].SupportedHardwareVersions.SupportedHardwareVersion {
+		if hwVersion.Name == name {
+			found = true
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("hardware version %s not found or not supported", name)
+	}
+
+	vdcHref, err := url.ParseRequestURI(vdc.Vdc.HREF)
+	if err != nil {
+		return nil, fmt.Errorf("error getting VDC href: %s", err)
+	}
+	vdcHref.Path += "/hwv/" + name
+
+	hardwareVersion := &types.VirtualHardwareVersion{}
+
+	_, err = vdc.client.ExecuteRequest(vdcHref.String(), http.MethodGet, types.MimeVirtualHardwareVersion, "error getting hardware version: %s", nil, hardwareVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	return hardwareVersion, nil
+}
+
+// Get highest supported hardware version of a VDC
+func (vdc *Vdc) GetHighestHardwareVersion() (*types.VirtualHardwareVersion, error) {
+	err := vdc.Refresh()
+	if err != nil {
+		return nil, err
+	}
+
+	hardwareVersions := vdc.Vdc.Capabilities[0].SupportedHardwareVersions.SupportedHardwareVersion
+	// Get last item (highest version) of SupportedHardwareVersions
```

Comment:
How do we know that the last one is the highest?

Reply:
The API returns it in that way in the versions I observed, but it doesn't return any timestamp which might explain the sort. e.g SupportedHardwareVersion JSON struct 
```json
{
	"_type": "SupportedHardwareVersionType",
	"content": "vmx-18",
	"name": "vmx-18",
	"href": "<<vcdUrl>>/api/vdc/f391a09a-3635-43f4-8d27-f607882d17d1/hwv/vmx-18",
	"type": "application/vnd.vmware.vcloud.virtualHardwareVersion+xml",
	"default": null
}
```